client/webserver/webserver.go

```diff
@@ -268,6 +297,39 @@ func New(cfg *Config) (*WebServer, error) {
 		WriteTimeout: 2 * time.Minute,                      // request to response time, must be long enough for slow handlers
 	}
 
+	if cfg.CertFile != "" || cfg.KeyFile != "" {
+		// Find or create the key pair.
+		keyExists := dex.FileExists(cfg.KeyFile)
+		certExists := dex.FileExists(cfg.CertFile)
+		if certExists != keyExists {
+			return nil, fmt.Errorf("missing cert pair file")
+		}
+		if !keyExists {
+			if err := genCertPair(cfg.CertFile, cfg.KeyFile, []string{cfg.Addr}); err != nil {
+				return nil, err
+			}
+		}
+		keyPair, err := tls.LoadX509KeyPair(cfg.CertFile, cfg.KeyFile)
+		if err != nil {
+			return nil, err
+		}
+		httpServer.TLSConfig = &tls.Config{
+			ServerName:               cfg.Addr,
+			Certificates:             []tls.Certificate{keyPair},
+			CurvePreferences:         []tls.CurveID{tls.CurveP521, tls.CurveP384, tls.CurveP256},
+			PreferServerCipherSuites: true,
+			CipherSuites: []uint16{
+				tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+				tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+				tls.TLS_RSA_WITH_AES_256_GCM_SHA384,
+				tls.TLS_RSA_WITH_AES_256_CBC_SHA,
+			},
```

Comment:
Can you comment on where these choices come from?

Reply:
Supposedly how you get a perfect ssl labs score.  https://github.com/scusi/PerfectSSLScore/blob/master/main.go
Probably too restrictive though.